src/hooks/stations.ts

```diff
@@ -48,10 +48,27 @@ export const useBooking = () => {
         }),
     [BUCHUNGEN],
   );
+  const refreshBooking = useCallback(
+    async () => {
+      const oldBooking = await getCurrentBooking();
+      return cancelBooking()
+        .then(() => {
+          if (oldBooking) {
```

Comment:
Hmm I think this should still try to book even if the booking has gone away. It just needs to be aware of the case that the re-book in that case can fail because there are no bookable bikes anymore.

Reply:
It only fails when there isn't a booking at the time of pressing the button. It can expire after `await getCurrentBooking()` without any problems.

When it fails however, there isn't any data available on which station to book. The only option would be the currently selected Station, but since that is implemented as state in `bike-map.tsx`, I can't see how to access it in `station.ts` without a lot of work.

Reply:
But when the function is called there _must_ be a booking stored in the hook (in the state variable), so we definitely know which station the booking was at.

Reply:
a61647b0581e23cac71344be89b16cb384708e83 Ok, so i simply took the older `booking` value from the state if fetching the current one returns a falsy value.